geles-frontend/src/EditOrder.svelte

```diff
@@ -0,0 +1,517 @@
+<script>
+  import { onMount } from "svelte";
+  import { user } from "./stores";
+  import { server_url } from "./index";
+  import axios from "axios";
+  import { getContext } from "svelte";
+  import { notificationContextKey } from "./contexts";
+
+  import { Link } from "svelte-routing";
+  import { bind, text } from "svelte/internal";
+  import { OrderStatus } from "./enums";
+  import { AppNotificationType } from "./enums";
+
+  const { addLoadingNotification, addNotification } =
+    getContext<AppNotificationContext>(notificationContextKey);
+
+  export let id: string;
+  $: isAdmin = $user && $user.admin;
+
+  interface OrderEdit {
+    orderFlowers: OrderFlower[];
+    address: string;
+    contactPhone: string;
+    version: number;
+  }
+
+  enum OrderEditButton {
+    SAVE,
+    PAY,
+    CANCEL,
+    ADMIN_CONFIRMPAY,
+    ADMIN_CONFIRMDELIVERED
+  }
+
+  let order: Omit<Order, "id" | "userId"> = {
+    createdDate: "",
+    totalOrderPrice: 0,
+    address: "",
+    contactPhone: "",
+    orderFlowers: [],
+    orderStatus: OrderStatus.CANCELED,
+    version: 0
+  };
+
+  let editDto: OrderEdit = {
+    orderFlowers: [],
+    contactPhone: "",
+    address: "",
+    version: 0
+  };
+
+  let editedUser: User = {
+    id: 0,
+    username: "",
+    photo: null,
+    admin: false,
+    cartId: 0
+  };
```

Comment:
Keep `let editedUser: User;` or `let editedUserName: string;`, since it's the only piece of data of user info used, it's safe to discard the rest